source/common/tracing/http_tracer_impl.h

```diff
@@ -190,15 +96,17 @@ class HttpNullTracer : public HttpTracer {
 
 class HttpTracerImpl : public HttpTracer {
 public:
-  HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& local_info);
+  HttpTracerImpl(DriverSharedPtr driver, const LocalInfo::LocalInfo& local_info);
 
   // Tracing::HttpTracer
   SpanPtr startSpan(const Config& config, Http::RequestHeaderMap& request_headers,
                     const StreamInfo::StreamInfo& stream_info,
                     const Tracing::Decision tracing_decision) override;
 
+  DriverSharedPtr driverForTest() const { return driver_; }
+
 private:
-  DriverPtr driver_;
+  DriverSharedPtr driver_;
```

Comment:
Yep. This is a long-standing problem and I hope it can be resolved in the future.

```
  /**
   * Create a particular tracer driver implementation. If the implementation is unable to produce
   * an tracer driver with the provided parameters, it should throw an EnvoyException in the case
   * of general error or a Json::Exception if the json configuration is erroneous. The returned
   * pointer should always be valid.
   *
   * NOTE: Due to the corner case of OpenCensus, who can only support a single tracingYaroslav Skopets, 1 year ago: • tracing: ensure that OpenCensus is configured…
   *       configuration per entire process, the returned Driver instance is not guaranteed
   *       to be unique.
   *       That is why the return type has been changed to std::shared_ptr<> instead of a more
   *       idiomatic std::unique_ptr<>.
   *
   * @param config supplies the proto configuration for the HttpTracer
   * @param context supplies the factory context
   */
  virtual Tracing::DriverSharedPtr createTracerDriver(const Protobuf::Message& config,
                                                      TracerFactoryContext& context) PURE;
```

Reply:
I see. Thanks!